### PR TITLE
rsyslog - fix retry logic - use local vars in templates

### PR DESCRIPTION
**Pull request by richm**
When retrying, the _id and _index fields should already be
set - use them - otherwise, create new values for them.
Also fixes tabs/spaces.

### CODE REVIEW

File: hack/testing/rsyslog/elasticsearch.conf.j2 [hunk 1]
```diff
@@ -11,33 +11,44 @@ template(name="index_pattern" type="list") {
...
+        }
+        # else fall through to retry operation
+    }
+    if strlen($!viaq_msg_id) > 0 then {
+	    set $.viaq_msg_id = $!viaq_msg_id;
+    } else if strlen($.omes!_id) > 0 then {
+        # retry
+        set $.viaq_msg_id = $.omes!_id;
+    } else {
+	    # NOTE: depends on rsyslog being compiled with --enable-uuid
```
Comment (nhosoi) [inline hack/testing/rsyslog/elasticsearch.conf.j2]: Where --enable-uuid is set?  I cannot find it in the spec files...
An indentation issue in line 43? ;)

Comment (richm) [reply hack/testing/rsyslog/elasticsearch.conf.j2]: > Where --enable-uuid is set? I cannot find it in the spec files...

It is set by default:
```
>./configure --help|grep uuid
  --enable-uuid           Enable support for uuid generation [default=yes]
```

fixed the indentation issue

Comment (richm) [reply hack/testing/rsyslog/elasticsearch.conf.j2]: see also https://kojipkgs.fedoraproject.org//packages/rsyslog/8.36.0/1.fc28/data/logs/x86_64/build.log
```
rsyslog will be compiled with the following settings:
...
    uuid support enabled:                     yes
```

Comment (nhosoi) [reply hack/testing/rsyslog/elasticsearch.conf.j2]: I see.  If it's set by default, it's even better.
/lgtm

**Pull request merged**